components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -71,6 +71,35 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         break
       }
 
+      if (tab && tab.url && tab.url.includes('twitch.tv')) {
+        let url = tab.url
+        // use an interval here to monitor when the DOM has finished
+        // generating. clear after the data is present.
+        // Check every 3 seconds for 1 minute
+        const markupMatch = '<figure class=\"tw-avatar tw-avatar--size-36\">' +
+                            '<div class=\"tw-border-radius-medium tw-overflow-hidden\">' +
+                            '<img class=\"tw-avatar__img tw-image\" alt=\"'
+        let itr = 0
+        const limit = 20
+        let interval = setInterval(function () {
```

Comment:
does this effect responsiveness of twitch in anyway?

Reply:
who clears interval is you close panel?

Reply:
internal is cleared at `limit` still if panel is closed. Although it still clears it can be restarted by opening the panel again. I would think we could set the limit down to 4 or 5 (vice 20). wdyt?

Even under a development run I do not notice any considerable difference in performance when interacting with the site. This is a small line of eval code that returns the outerHtml that, when generated with the parts we need, equates to roughly 220Kb

Reply:
can we clear interval as soon as you close panel?

Reply:
There may be a way but it would involve setting a local storage flag whenever panel is opened and then resetting it when it closes. 

example:
```
chrome.runtime.onConnect.addListener(function (externalPort) {
  chrome.storage.local.set({
    'rewards_panel_open': true
  })
  externalPort.onDisconnect.addListener(function () {
    chrome.storage.local.set({
      'rewards_panel_open': false
    })
  })
})
```
I'm in the process of testing this now but do you see any immediate problems that could lead to inconsistencies?

Reply:
In addition, we seem to get the information we need right away, so I would also want to shorten the limit as a safeguard fallback

Reply:
So using the above, we are able to clear the timeout when the panel is closed. Test is successful.